CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/icon/IconDataNGTest.java

```diff
@@ -0,0 +1,540 @@
+        final int result6 = scaledImage6.getWidth() * scaledImage6.getHeight();
+
+        assertEquals(result6, expResult6);
+    }
+
+    /**
+     * Test of scaleImage method, of class IconData, when image is null.
+     */
+    @Test
+    public void testScaleImage_imageIsNull() {
+        System.out.println("testScaleImage_imageIsNull");
+
+        final BufferedImage image = null;
+        final int size = 2;
+
+        iconDataStaticMock.when(() -> IconData.scaleImage(Mockito.eq(image), Mockito.eq(size))).thenCallRealMethod();
+        final BufferedImage scaledImage = IconData.scaleImage(image, size);
+
+        final BufferedImage expResult = null;
+        final BufferedImage result = scaledImage;
+
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test of scaleImage method, of class IconData, when image type not
+     * TYPE_4BYTE_ABGR.
+     */
+    @Test
+    public void testScaleImage_imageNotTYPE_4BYTE_ABGR() {
+        System.out.println("testScaleImage_imageNotTYPE_4BYTE_ABGR");
+
+        final BufferedImage image = new BufferedImage(1, 1, BufferedImage.TYPE_3BYTE_BGR);
+        final int size = 2;
+
+        iconDataStaticMock.when(() -> IconData.scaleImage(Mockito.eq(image), Mockito.eq(size))).thenCallRealMethod();
+        final BufferedImage scaledImage = IconData.scaleImage(image, size);
+
+        final int expResult = BufferedImage.TYPE_4BYTE_ABGR;
+        final int result = scaledImage.getType();
+
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test of equals method, of class IconData.
+     */
+    @Test
+    public void testEquals() {
+        System.out.println("testEquals");
+
+        final IconData instance = new IconDataImpl();
+
+        // When obj is null.
+        Object obj = null;
+        assertFalse(instance.equals(obj));
+
+        // When the class of obj is not the same class as this.
+        obj = new Object();
+        assertFalse(instance.equals(obj));
+
+        // When the class of obj is the same class as this but has a different value for data.
+        obj = new IconDataImpl();
+
+        instance.setData(new byte[1]);
+        ((IconData) obj).setData(new byte[2]);
+
+        assertFalse(instance.equals(obj));
+
+        // When the class of obj is the same class as this and has the same value for data.
+        obj = new IconDataImpl();
+
+        instance.setData(new byte[1]);
+        ((IconData) obj).setData(new byte[1]);
+
+        assertTrue(instance.equals(obj));
+    }
+
+    /**
+     * Test of toString method, of class IconData.
+     */
+    @Test
+    public void testToString() {
+        System.out.println("testToString");
+
+        final IconData instance = new IconDataImpl();
+
+        final String expResult1 = String.format("Image of %d bytes", 0);
+        final String result1 = instance.toString();
+
+        assertEquals(result1, expResult1);
+
+        // When data is not null.
+        instance.setData(new byte[2]);
```

Comment:
could use `getData` with default values and if you don't want size 0 you could mock the value from `createData`